src/mesh_topology_analysis.rs

```diff
@@ -0,0 +1,313 @@
+use std::collections::HashMap;
+
+use smallvec::SmallVec;
+
+use crate::convert::{cast_u32, cast_usize};
+use crate::geometry::{Face, Geometry, UnorientedEdge};
+
+/// Calculates topological relations (neighborhood) of mesh face -> faces.
+/// Returns a Map (key: face index, value: list of its neighboring faces indices)
+#[allow(dead_code)]
+pub fn face_to_face_topology(geometry: &Geometry) -> HashMap<usize, SmallVec<[usize; 8]>> {
+    let mut f2f: HashMap<usize, SmallVec<[usize; 8]>> = HashMap::new();
+    for (from_counter, face) in geometry.faces().iter().enumerate() {
+        match face {
+            Face::Triangle(f) => {
+                let [f_e_0, f_e_1, f_e_2] = f.to_unoriented_edges();
+                for (to_counter, t_f) in geometry.triangle_faces_iter().enumerate() {
```

Comment:
Forgot to use `geometry.faces()` here instead of `geometry.triangle_faces_iter()`.

I have also created an issue to remove `geometry.triangle_faces_iter()` https://trello.com/c/vFi2zRtW/151-remove-geometrytrianglefacesiter